_docker/drupal/drupal-filesystem/web/modules/custom/rhd_disqus/src/Form/DisqusSettingsForm.php

```diff
@@ -64,6 +71,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $this->config('rhd_disqus.disqussettings')
       ->set('rhd_disqus_shortname', $form_state->getValue('rhd_disqus_shortname'))
       ->set('rhd_disqus_api_key', $form_state->getValue('rhd_disqus_api_key'))
+      ->set('rhd_disqus_secret_key', $form_state->getValue('rhd_disqus_secret_key'))
```

Comment:
@alistairmc I'd prefer us to provide the secret key via config overrides in each environment as this is how we provide all other secret keys to Drupal. The above approach causes us a problem as the production key will filter down into lower environments and it's also going to be stored unencrypted on disk by Drupal.

We can get it working on PR environment first as part of this. It's not a big job, I just need to know the key you want to use.

cheers,

Rob

Reply:
Sounds fine I only did it like this since we already had the public key and short name in the cms.
I have been using a personal Disqus account to test I would suggest that is not what we use moving forward for PR's. Whoever is an admin for the main Disqus account will need to create new API endpoints for the various environments to test with, all three values in the cms API public and private keys and short name need to be changed per environment unless we intend to just use one account. Its an architectural call on what we do so I guess up to you @robpblake

Reply:
@robpblake changed to using config, the for still displays the values in a read-only way for ease of identifying the values used, saving the form will not change the config.

I need info from @joeschram when he is back after July 4th for the keys for stage, pr and live.